mitiq/zne/tests/test_zne.py

```diff
@@ -560,3 +568,59 @@ def qs_noisy_simulation(
 
     mitigated = execute_with_zne(circuit, qs_noisy_simulation)
     assert abs(mitigated) < 1000
+
+
+@pytest.mark.parametrize(
+    "noise_scaling_method",
+    [fold_gates_at_random, insert_id_layers, fold_global, fold_all],
+)
+@pytest.mark.parametrize(
+    "extrapolation_factory", [RichardsonFactory, LinearFactory]
+)
+@pytest.mark.parametrize(
+    "to_frontend",
+    [None, to_qiskit, to_braket, to_pennylane, to_pyquil, to_qibo],
+)
+def test_two_stage_zne(
```

Comment:
There should be two separete unit tests for the two new functions you implemented. This is an integration test, and one bad symptom is that this test is violating the [AAA pattern](https://jamescooke.info/arrange-act-assert-pattern-for-python-developers.html).